window/src/os/windows/window.rs

```diff
@@ -716,72 +735,129 @@ impl WindowOps for Window {
-                    hdc,
-                    HP_HEADERITEM,
-                    HIS_NORMAL,
-                    TMT_CAPTIONFONT,
-                    &mut log_font,
-                );
-                if res == S_OK {
-                    CloseThemeData(theme);
-                    return Some(log_font);
-                }
-            }
+    fn get_os_parameters(&self, config: &ConfigHandle) -> anyhow::Result<Option<Parameters>> {
+        let hwnd = self.0 .0;
+        if hwnd.is_null() {
+            return Err(anyhow::anyhow!("HWND is null"));
+        }
 
-            let res = GetThemeSysFont(theme, TMT_CAPTIONFONT, &mut log_font);
-            if !theme.is_null() {
-                CloseThemeData(theme);
-            }
+        let has_focus = !unsafe { GetFocus() }.is_null();
+        let is_maximized = window_is_maximized(hwnd);
 
-            if res == S_OK {
-                Some(log_font)
+        let is_full_screen = unsafe {
+            let mut mi = MONITORINFO {
+                cbSize: std::mem::size_of::<MONITORINFO>() as u32,
+                ..Default::default()
+            };
+            if GetMonitorInfoW(MonitorFromWindow(hwnd, MONITOR_DEFAULTTOPRIMARY), &mut mi)
+                == winapi::shared::minwindef::TRUE
+            {
+                let mut client_rect = RECT::default();
+                if GetClientRect(hwnd, &mut client_rect) == winapi::shared::minwindef::TRUE {
+                    client_rect.right >= mi.rcMonitor.right - mi.rcMonitor.left
+                        && client_rect.bottom >= mi.rcMonitor.bottom - mi.rcMonitor.top
+                } else {
+                    false
+                }
             } else {
-                None
+                false
             }
-        }
-        unsafe {
-            let hwnd = self.0 .0;
+        };
+
+        let title_font = unsafe {
             let hdc = GetDC(hwnd);
-            let result = match populate_log_font(hwnd, hdc) {
-                Some(lf) => wezterm_font::locator::gdi::parse_log_font(&lf, hdc).ok(),
+            if hdc.is_null() {
+                return Err(anyhow::anyhow!("Could not get device context"));
+            }
+            let result = match get_title_log_font(hwnd, hdc) {
+                Some(lf) => Some(wezterm_font::locator::gdi::parse_log_font(&lf, hdc)?),
                 None => None,
             };
             ReleaseDC(hwnd, hdc);
             result
+        };
+
+        let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+        let use_accent = hkcu
+            .open_subkey("SOFTWARE\\Microsoft\\Windows\\DWM")?
+            .get_value::<u32, _>("ColorPrevalence")?;
+        let settings = UISettings::new()?;
+        let top_border_color = if has_focus {
+            if use_accent == 1 {
+                wuicolor_to_linearrgba(settings.GetColorValue(UIColorType::Accent)?)
+            } else {
+                if is_win10() {
+                    LinearRgba(0.01, 0.01, 0.01, 0.67)
```

Comment:
These hard coded numbers are a bit unfortunate.
Can we get the colors we need from https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getsystemmetrics instead?

Reply:
No, they aren't available there. Win32 was basically deprecated by Microsoft in favor of UWP, so lots of things there don't work on Windows 10+